src/javascript/JContent/ContentNavigation/ContentNavigation.container.jsx

```diff
@@ -1,47 +1,68 @@
 
-const ContentNavigationContainer = ({mode, siteKey, handleNavigation}) => {
-    let accordionItems = registry.find({type: 'accordionItem', target: 'jcontent'});
-
-    const currentState = useSelector(state => ({site: state.site, language: state.language}));
+const ContentNavigationContainer = ({handleNavigation, selectorObject, accordionItemTarget, accordionItemType, header, sitePermissions}) => {
+    const selector = useMemo(() => createStructuredSelector(selectorObject), [selectorObject]);
+    const dispatch = useDispatch();
+    const {siteKey, language, mode} = useSelector(state => selector(state));
     const permissions = useNodeChecks({
-        path: `/sites/${currentState.site}`,
-        language: currentState.language
+        path: `/sites/${siteKey}`,
+        language: language
     }, {
-        requiredSitePermission: [JContentConstants.accordionPermissions.pagesAccordionAccess, JContentConstants.accordionPermissions.contentFolderAccordionAccess, JContentConstants.accordionPermissions.mediaAccordionAccess, JContentConstants.accordionPermissions.additionalAccordionAccess, JContentConstants.accordionPermissions.formAccordionAccess]
+        requiredSitePermission: sitePermissions
     });
 
     if (permissions.loading) {
         return null;
     }
 
-    accordionItems = accordionItems.filter(accordionItem =>
+    let accordionItems = registry.find({type: accordionItemType, target: accordionItemTarget});
+    accordionItems = sitePermissions.length === 0 ? accordionItems : accordionItems.filter(accordionItem =>
         permissions.node && Object.prototype.hasOwnProperty.call(permissions.node.site, accordionItem.requiredSitePermission) && permissions.node.site[accordionItem.requiredSitePermission]
     );
 
-    return <ContentNavigation accordionItems={accordionItems} mode={mode} siteKey={siteKey} handleNavigation={handleNavigation}/>;
+    return (
+        <ContentNavigation header={header}
+                           accordionItems={accordionItems}
+                           mode={mode}
+                           siteKey={siteKey}
+                           handleNavigation={(mode, path) => dispatch(handleNavigation(mode, path))}
+        />
+    );
 };
 
-const mapDispatchToProps = dispatch => ({
-    handleNavigation: (mode, path) => dispatch(cmGoto({mode, path}))
-});
-
-let mapStateToProps = state => ({
-    mode: state.jcontent.mode,
-    siteKey: state.site
-});
-
 ContentNavigationContainer.propTypes = {
-    handleNavigation: PropTypes.func.isRequired,
-    mode: PropTypes.string.isRequired,
-    siteKey: PropTypes.string.isRequired
+    selectorObject: PropTypes.shape({
+        mode: PropTypes.func.isRequired,
+        siteKey: PropTypes.func.isRequired,
+        language: PropTypes.func.isRequired
+    }),
+    accordionItemTarget: PropTypes.string,
+    accordionItemType: PropTypes.string,
+    handleNavigation: PropTypes.func,
```

Comment:
Maybe rename to `handleNavigationAction`  to understand we are passing a redux action